frontend/components/common/FlexScrollGradient.tsx

```diff
@@ -76,10 +76,7 @@ function getGradientsBaseStyle(
    };
 }
 
-function getGradientStyleProps(
-   el: HTMLElement,
-   gradientSizePX: number
-): InnerFlexStyling {
+function getGradientStyleProps(el: HTMLElement): InnerFlexStyling {
```

Comment:
@danielbeardsley I'd love to continue our convo from https://github.com/iFixit/react-commerce/pull/1923#discussion_r1301954469

Basically, your concern was that this component was complex. I agree.
The component has a few jobs and we might be able to reduce the work it is doing.

Before getting into that, I wanted to comment that I'd love a pure CSS solution to this, but there is not an exist CSS feature to allow CSS only when scrolled, not at either end, but [they are working on it](https://daverupert.com/2023/08/animation-timeline-scroll-shadows/)


Here's the jobs of this component:

* Add a before / after element iff scrolled
* Change the opacity of the before/after while scrolling
	* Before hidden at start, after hidden when scrolled fully
* Adjust for the size of the scrollbar
	* Say content is 100px, but then you add scroll auto, the browser adds a scrollbar (chrome's is 15px) and not 'width: 100%' will be `115px` Given the positioning of the fades, they would overlay the scrollbar, thus we must subtract the scrollbar
* Handle both row and column flex directions
	* All math is the same but inversed
* Adjust the size of the fade based on the available space of the scroll container

Reply:
> Basically, your concern was that this component was complex. I agree.
> ... Before getting into that, I wanted to comment that I'd love a pure CSS solution to this

I don't mean to downplay what you done, it's pretty snazzy, but I tend to think it's not a problem that needs solving. It seems a bit of a stretch to effectively say "native scrolling has a bad UI that we need to fix". Every website deals with scrolling, what makes this particular scroll area unique such that it needs a solution?

Reply:
> It seems a bit of a stretch to effectively say "native scrolling has a bad UI that we need to fix".

Totally, the main reason this exists is that we keep designing fades (usually for mobile) so we indicate to a user that a section is scrollable.

Some examples:
![image](https://github.com/iFixit/react-commerce/assets/857362/67db9ff0-c43f-482a-b5b9-7c40f5212326)
![image](https://github.com/iFixit/react-commerce/assets/857362/e292c542-55ca-45ed-a535-e8188469473c)
![image](https://github.com/iFixit/react-commerce/assets/857362/9d083f63-3888-4778-b2c2-1d7329c263a6)

This seems to be part of our new design language, but if we decide we don't need it / can simplify that would be great.

Reply:
I think augmenting native scrolling with fades is fairly complex and really easy to mess up, potentially breaking the scroll behavior. I vote we should rely on the presence of scroll bars and the specifying a height that isn't an exact multiple of the items (so one is chopped off).

That said, I think region vertical scrolling on mobile devices is a frustrating experience because it breaks the "drag anywhere to scroll the screen" model that we're all so used to.

Thoughts @ianrohde and @federicobadini and @mmarcon91 ?

Reply:
Overall I agree with you Danny, but I want to point out one thing.

> I vote we should rely on the presence of scroll bars and then specifying a height that isn't an exact multiple of the items (so one is chopped off).

This isn't always possible. For example, the guide flag text is translatable and the adding/removing of flags is user-editable. It would be possible for some combination of strings/flags to be near pixel-perfect to the edge of the scrollable pane. Without the fade, it would not seem scrollable at a glance.

Similarly, I suspect the font stack could wrap text differently and have similar issues.

---

I just wanted to claim that your statement is totally reasonable, but in practice may not be possible due to many variables.

Reply:
In specific, I think the little drop down is not that useful on mobile. I think a "Scroll to Table of Contents" link would be more friendly and much easier to implement.

Reply:
The TOC matches the Figma comp, but Jarred went the extra mile to add the visual cue that the list is scrollable:

<details>

<img width="342" alt="Screenshot 2023-08-29 at 11 55 32 AM" src="https://github.com/iFixit/react-commerce/assets/1634505/11f1156b-5c3c-440c-9179-3ade5acb1837">

</details>

> That said, I think region vertical scrolling on mobile devices is a frustrating experience because it breaks the "drag anywhere to scroll the screen" model that we're all so used to.

I'm betting that Mattia considered this, but it's just difficult to convey in Figma.

Reply:
> I'm betting that Mattia considered this, but it's just difficult to convey in Figma.

There is a fade in the demo app he built on both desktop but not mobile: https://rwddn4.csb.app/

I think I had assumed the fade was intended and I had already invested time in this component from the other repo and new it was easy enough to drop in.

If not intended, I can easily remove it.

Reply:
I think fades are good, especially for horizontal scrolls. There are rarely any permanently visible scroll bars on mobile, so I don't think that's a good answer here. The fade on a single line of text seems a little hard to understand to me, so maybe it just needs to be a shorter gradient.

Alternatively, it seems like this dropdown could just be a bigger, maybe even full-screen on mobile since dropdowns are a little funky even in the best case.

Reply:
For me we can remove the shading in the mobile summary panel. They make reading complicated and reduce the visible space.

We can also keep the native system scrolling and remove the bars, in 90% of cases the point-title at the bottom of the panel will be cut off and the user should understand that there is content to scroll through.

Reply:
In the prototype

Reply:
In the prototype, I've implemented the fade with Tailwind like this:

```html
<div class="h-24 w-full bottom-0 absolute left-0 bg-gradient-to-t from-gray-50 pointer-events-none"></div>
```

It should be possible to replicate this using Charka, right?

Reply:
@mmarcon91 im not quite understanding what you mean with you comment.

The code above is effectively replicating the fade you described. The only difference is the height varies based on the available space. This was done so that the fade component can be used anywhere on any design

Reply:
Not necessarily the fade height should vary based on the container height. The fades that I design are based on the specific context, they don't follow some specific mathematic rules, so I don't think it's really possible to express this as a reusable component. 
We would also probably benefit in terms of performance from using a css only solution, what do you think?

Reply:
> We would also probably benefit in terms of performance from using a css only solution, what do you think?

Totally, but a CSS-only solution does not work how you might expect.

There is no CSS hook to determine if you have scrolled to the top or bottom. There is also no CSS to decide whether your element is scrollable at all.

You can totally have a fade that is always there, but floated over nothing like your example:
![image](https://github.com/iFixit/react-commerce/assets/857362/67470fd7-8e1f-43f3-8884-3291a7e3be3f)


But if we want to only add the scroll if and only if the container is scrollable we need a use effect.

@ianrohde and I have taken some time to look at a CSS-only solution and we still lean towards the current solution for now.